namespace_registry/registry_db_test.go

```diff
@@ -0,0 +1,181 @@
+/***************************************************************
+ *
+ * Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+package nsregistry
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCacheAdminNoApproval(t *testing.T) {
+	viper.Reset()
+	registry_db_dir := t.TempDir()
+
+	viper.Set("Registry.DbLocation", registry_db_dir)
+
+	err := InitializeDB()
+	assert.NoError(t, err, "error initializing registry database")
```

Comment:
So, I looked this up as it seems to be personal preference. There is a minor difference. Basically, `require` will cause a test to quit and exit immediately, which may be what we want more often, but with go routines, it can cause some issues.

To quote someone else:
"However, if the test has goroutines, we need to be careful that the test waits for goroutines' completion. Otherwise, the test errors get buried/missed; leaving you wondering why a test case failed!"